src/src/rx-serial/SerialDisplayport.cpp

```diff
@@ -57,4 +59,46 @@ uint32_t SerialDisplayport::sendRCFrame(bool frameAvailable, bool frameMissed, u
     send(MSP_STATUS, &status_DJI, sizeof(status_DJI), _outputPort);
 
     return MSP_MSG_PERIOD_MS;   // Send MSP msgs to DJI at 10Hz
-}
\ No newline at end of file
+}
+
+void SerialDisplayport::processBytes(uint8_t *bytes, u_int16_t size)
+{
+    // Super-basic air-unit detection:
+    // Wait for at least 6 bytes to be received (minimum MSP msg length)
+    // before we decide we are connected to a DJI air unit
+    if (m_receivedBytes < 6)
+    {
+      m_receivedBytes += size;
+    }
+}
+
+bool SerialDisplayport::getArmedState(uint32_t *channelData)
+{
+    if (firmwareOptions.dji_permanently_armed)
+    {
+        // If we are using permanent arming then we need to make sure the air-unti is connected
+        // The O3 needs to see the arm state change (i.e. false --> true) after is has fully booted
+        // Wait for activity on the UART from the air-unit, then wait 10 seconds before arming
+
+        if (m_receivedBytes >= 6)
+        {
+            // Start the timer if it's the first time receiving 6 or more bytes
+            if (m_receivedTimestamp == 0)
+            {
+                m_receivedTimestamp = millis();
+            }
+            // Arm permanently after 10 seconds
+            return millis() > m_receivedTimestamp + 10000;
+        }
+
+        return false;
+    }
+    else
+    {
+        // If we are not using permanent arming then we don't need to wait for the air-unit to be connected
+        // The arm channel will provide the required state change to arm the O3
+        return channelData[4] > CRSF_CHANNEL_VALUE_MID;
```

Comment:
I merged #3008. Please include `OTA.h `and change `return channelData[4] > CRSF_CHANNEL_VALUE_MID;` to `return isArmed;`

isArmed provides a generalized armed status regardless of the Arming method (CH5 and also for the added optional Arming method Switch). For more context see #3008.